### PR TITLE
add Fluxcd addon 

### PR DESCRIPTION
**Pull request by yangsoon**
**What this PR does / why we need it**:

refer to https://github.com/oam-dev/kubevela/issues/1760#issuecomment-868940314, add fluxcd as a built-in initializer. and add a `gitops` component(refer to https://github.com/oam-dev/kubevela/issues/1562).

**Which issue(s) this PR fixes**:

Fixes https://github.com/oam-dev/kubevela/issues/1562

**Special notes for your reviewer**:

1. add `fluxcd` related files at `./vela-system/addons/`.
2. `./vela-system/gen_addons.go` can help collect related resources of addon to configmap(this can be used by vela-cli).

### CODE REVIEW

File: vela-templates/registry/auto-gen/gitops.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        		validation: "client"
+        	}
+        }
+        
+        parameter: {
+        	//+usage=The repository URL, can be a HTTP/S or SSH address.
+        	repoUrl: string
+        
+        	//+usage=The interval at which to check for repository updates.
+        	interval: *"1m" | string
```
Comment (wonderflow) [inline vela-templates/registry/auto-gen/gitops.yaml:52]:
```suggestion
        	pull_interval: *"1m" | string
```

File: vela-templates/registry/auto-gen/gitops.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Code generated by KubeVela templates. DO NOT EDIT.
+apiVersion: core.oam.dev/v1beta1
+kind: ComponentDefinition
+metadata:
+  name: gitops
```
Comment (wonderflow) [inline vela-templates/registry/auto-gen/gitops.yaml:5]:
```suggestion
  name: kustomize
```

File: vela-templates/registry/auto-gen/gitops.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        	repoUrl: string
+        
+        	//+usage=The interval at which to check for repository updates.
+        	interval: *"1m" | string
+        
+        	//+usage=The Git reference to checkout and monitor for changes, defaults to master branch.
+        	branch: *"master" | string
+        
+        	//+usage=Path to the directory containing the kustomization.yaml file, or the set of plain YAMLs a kustomization.yaml should be generated for.
+        	path: string
```
Comment (wonderflow) [inline vela-templates/registry/auto-gen/gitops.yaml:58]: can we specify many path

File: vela-templates/registry/auto-gen/gitops.yaml (hunk 2 above)
Comment (wonderflow) [reply vela-templates/registry/auto-gen/gitops.yaml:5]: 做到 built-in

File: vela-templates/registry/auto-gen/gitops.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+metadata:
+  name: gitops
+  namespace: vela-system
+  annotations:
+    definition.oam.dev/description: "gitops can fetching, building, updating and applying Kustomize manifests from git repo."
+spec:
+  workload:
+    definition:
+      apiVersion: source.toolkit.fluxcd.io/v1beta1
+      kind: GitRepository
```
Comment (wonderflow) [inline vela-templates/registry/auto-gen/gitops.yaml:13]: 这里应该是不填

File: vela-templates/registry/auto-gen/gitops.yaml (hunk 3 above)
Comment (yangsoon) [reply vela-templates/registry/auto-gen/gitops.yaml:56]: kustomize CRD only provide one path to set the directory containing the kustomization.yaml file
https://fluxcd.io/docs/components/kustomize/kustomization/

File: Makefile [hunk 5]
```diff
@@ -232,6 +232,7 @@ manifests: installcue kustomize
 	mv config/crd/base/* charts/vela-core/crds
 	./vela-templates/gen_definitions.sh
 	./hack/crd/cleanup.sh
+	go run ./vela-templates/gen_addons.go --addons-path=./vela-templates/addons --store-path=./charts/vela-core/templates/addons
```
Comment (wonderflow) [inline Makefile:235]: 这个工具应该从纯 CRD 对象生成出 ComponentDefinition 和 Initializer，而不是纯粹把写好的ComponentDefinition文件合并一下。

**Review by wonderflow — APPROVED** [review]

File: Makefile (hunk 5 above)
Comment (yangsoon) [reply Makefile:238]: got it

File: charts/vela-core/templates/defwithtemplate/raw.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# Code generated by KubeVela templates. DO NOT EDIT.
+apiVersion: core.oam.dev/v1beta1
+kind: ComponentDefinition
+metadata:
+  annotations:
+    definition.oam.dev/description: ""
```
Comment (wonderflow) [inline charts/vela-core/templates/defwithtemplate/raw.yaml:6]: raw allow users to specify raw K8s object in properties

File: charts/vela-core/templates/defwithtemplate/raw.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  name: raw
+  namespace: vela-system
+spec:
+  workload:
+    type:
```
Comment (wonderflow) [inline charts/vela-core/templates/defwithtemplate/raw.yaml:11]: 填上

**Pull request merged**